src/main/java/com/opentable/kafka/builders/KafkaConsumerBuilder.java

```diff
@@ -9,66 +14,111 @@
+    public KafkaConsumerBuilder<K, V> withLoggingSampleRate(double rate) {
+        kafkaBaseBuilder.loggingSampleRate = rate;
         return this;
     }
 
-    public KafkaConsumerBuilder<K, V> withLoggingSampleRate(Double rate) {
-        return withProp(LoggingInterceptorConfig.SAMPLE_RATE_PCT_CONFIG, rate);
-    }
-
     public KafkaConsumerBuilder<K, V> withInterceptor(Class<? extends ConsumerInterceptor<K, V>> clazz) {
-        setListPropItem(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, clazz.getName());
+        kafkaBaseBuilder.interceptors.add(clazz.getName());
         return this;
     }
 
     public KafkaConsumerBuilder<K, V> withGroupId(String val) {
-        return withProp(ConsumerConfig.GROUP_ID_CONFIG, val);
+        groupId = Optional.ofNullable(val);
+        return this;
     }
 
     public KafkaConsumerBuilder<K, V> withOffsetReset(AutoOffsetResetType val) {
-        return withProp(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, val.value);
+        autoOffsetResetType = val == null ? AutoOffsetResetType.None : autoOffsetResetType;
+        return this;
     }
 
     public KafkaConsumerBuilder<K, V> withMaxPollRecords(int val) {
-        return withProp(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, val);
+        maxPollRecords = Optional.of(val);
+        return this;
     }
 
-    public <K2, V2> KafkaConsumerBuilder<K2, V2> withDeserializers(Class<? extends Deserializer<K2>> keyDeSer, Class<? extends Deserializer<V2>> valDeSer) {
-        prop.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeSer);
-        prop.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valDeSer);
-        return new KafkaConsumerBuilder<>(prop);
+    public KafkaConsumerBuilder<K, V> withDeserializers(Class<? extends Deserializer<K>> keyDeSer, Class<? extends Deserializer<V>> valDeSer) {
+        this.keyDe = keyDeSer;
+        this.valueDe = valDeSer;
+        return this;
     }
 
+    public KafkaConsumerBuilder<K, V> withBootstrapServer(String bootStrapServer) {
+        kafkaBaseBuilder.withBootstrapServer(bootStrapServer);
+        return this;
+    }
+
+    public KafkaConsumerBuilder<K, V> withBootstrapServers(List<String> bootStrapServers) {
+        kafkaBaseBuilder.withBootstrapServers(bootStrapServers);
+        return this;
+    }
+
+    public KafkaConsumerBuilder<K, V> withClientId(String val) {
+        kafkaBaseBuilder.withClientId(val);
+        return this;
+    }
+
+    public KafkaConsumerBuilder<K, V> withSecurityProtocol(String protocol) {
+        kafkaBaseBuilder.withSecurityProtocol(protocol);
+        return this;
+    }
+
+    public KafkaConsumerBuilder<K, V> withMetricRegistry(MetricRegistry metricRegistry) {
+        kafkaBaseBuilder.withMetricRegistry(metricRegistry);
+        return this;
+    }
+
+
     public KafkaConsumer<K, V> build() {
-        // TODO: add checks here
-        return new KafkaConsumer<>(prop);
+        kafkaBaseBuilder.baseBuild();
```

Comment:
Nothing is combined into props (except the base props, and any added base props) until now